confdgnmi/testtool/README.md

```diff
@@ -0,0 +1,122 @@
+It uses a [Robot Framework](https://robotframework.org/) (a generic open source automation framework) to implement range of test suites and
+test cases. These try to identify capabilities of a device to service typical telemetry related gNMI requests.
+
+## requirements
+
+We strongly depend on **Python3** for running the tests. All the `python` commands in following text assume that this binary resolves to a (fairly recent) python3 interpreter. Following is an example of supported (not necessarily lowest required) version:
+
+```bash
+$ python --version
+Python 3.8.10
+```
+
+See Robot's [installation instructions](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#installation-instructions) for details on how to install the framework. To verify you have a `robot` installed, you can run `robot --version` shell command. As of writing this document, current version is:
+
+```bash
+$ robot --version
+Robot Framework 6.0.2 (Python 3.8.10 on linux)
+```
+
+Actual test-runs require that supplementary gNMI client code (located in `../src`, primarily `confd_gnmi_client.py` module) can be run successfully. See the `../README.md` for requirements if you run into some issues.
+
+## generating test specification
+
+You can generate the "specification" of all the test suites/test cases from
+the existing robot files without need of a live target device to test.
+You can do this using Robot's supporting tool called [testdoc](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#test-data-documentation-tool-testdoc).
+
+Assuming you have all the [requirements](#requirements) installed, just execute following shell command from within the current working directory (the one where this README.md file is located):
+
+```bash
+python -m robot.testdoc ./ TestDocumentation.html
+```
+
+This creates a single HTML file with specified name - that includes descriptions and details of all the test suites and test-cases from the project's `.robot` files.
+
+It has a structure of a packed tree nodes that you can open to see specific test-suites and/or test-cases hierarchy, and read their names, descriptions, etc.
+
+If needed, see "`python -m robot.testdoc --help`" for possible output configuration.
+
+## running the tests
+
+To run any tests, we you need a target device configuration file first.
+
+When a test run is executed and finished, it creates few files by default (unless modified by `robot` command parameters).
+- `report.html` - good as a starting point, summary of all the tests suites/cases and their results;<br/>
+    it links into the following `log.html` file for details...
+- `log.html` - detailed log for all the steps executed in test run
+
+### PYTHONPATH and executing tests
+
+Whether you run tests against "real" device, or the "ConfD adapter" (see further), you need properly set Python environment for tests to execute correctly.
```

Comment:
Isn't "You need properly set Python environment for tests to execute correctly" enough?